### PR TITLE
[Fix](inverted index) if range query exceeds CLucene limits, downgrade it from inverted index

### PR DESCRIPTION
**Pull request by airborne12**
## Proposed changes

Issue Number: close #xxx

CLucene may throw CL_ERR_TooManyClauses when a range query hits too many terms.
In this situation, we have to downgrade from inverted index.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/inverted_index_reader.cpp [hunk 1]
```diff
@@ -491,8 +499,15 @@ Status StringTypeInvertedIndexReader::query(OlapReaderStatistics* stats,
                                     result.add(docid);
                                 });
     } catch (const CLuceneError& e) {
-        LOG(WARNING) << "CLuceneError occured, error msg: " << e.what();
-        return Status::Error<ErrorCode::INVERTED_INDEX_CLUCENE_ERROR>();
+        if (_is_range_query(query_type) && e.number() == CL_ERR_TooManyClauses) {
+            LOG(INFO) << "range query term exceeds limits, try to downgrade from inverted index,"
+                      << "column name:" << column_name << " search_str:" << search_str;
+            return Status::Error<ErrorCode::INVERTED_INDEX_FILE_HIT_LIMIT>();
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index_reader.cpp:505]: suggest a more generic name INVERTED_INDEX_IGNORE_INDEX

File: be/src/olap/rowset/segment_v2/inverted_index_reader.cpp [hunk 2]
```diff
@@ -491,8 +499,15 @@ Status StringTypeInvertedIndexReader::query(OlapReaderStatistics* stats,
                                     result.add(docid);
                                 });
     } catch (const CLuceneError& e) {
-        LOG(WARNING) << "CLuceneError occured, error msg: " << e.what();
-        return Status::Error<ErrorCode::INVERTED_INDEX_CLUCENE_ERROR>();
+        if (_is_range_query(query_type) && e.number() == CL_ERR_TooManyClauses) {
+            LOG(INFO) << "range query term exceeds limits, try to downgrade from inverted index,"
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index_reader.cpp:503]: WARNING

File: be/src/olap/rowset/segment_v2/inverted_index_reader.cpp (hunk 1 above)
Comment (airborne12) [reply be/src/olap/rowset/segment_v2/inverted_index_reader.cpp:505]: may be INVERTED_INDEX_BYPASS better?

File: be/src/olap/rowset/segment_v2/inverted_index_reader.cpp [hunk 3]
```diff
@@ -73,6 +73,14 @@
 namespace doris {
 namespace segment_v2 {
 
+bool InvertedIndexReader::_is_range_query(InvertedIndexQueryType query_type) {
+    return (query_type == InvertedIndexQueryType::EQUAL_QUERY ||
```
[the remark this answers is not in the document]
Comment (airborne12) [reply be/src/olap/rowset/segment_v2/inverted_index_reader.cpp:77]: will fix that

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**